check_ssl.py

```
@@ -0,0 +1,118 @@
+import subprocess
+import smtplib
+from uuid import uuid4
+import inspect
+from datetime import datetime, timezone
+from unittest.mock import patch
+
+# Send email
+def send_email(email_host, email_port, user, password, email_from, email_to, email_body, email_subject):
+    
+    _user, email_domain = email_from.split('@', 1)
+    email_text = inspect.cleandoc(f"""\
+    Message-ID: <{uuid4()}@{email_domain}>
+    From: Python SMTP Sender <%s>
+    To: %s
+    Subject: %s
+
+    %s
+    """) % (email_from, email_to, email_subject, email_body)
+    try:
+        smtp = smtplib.SMTP_SSL if email_port == 465 else smtplib.SMTP
+        smtp_server = smtp(email_host, email_port)
+        smtp_server.ehlo()
+
+        if email_port == 465:
+            smtp_server.login(user, password)
+
+        smtp_server.sendmail(email_from, email_to, email_text.encode('UTF-8'))
+        smtp_server.close()
+        print ("Email sent successfully!")
+    except Exception as ex:
+        print ("Something went wrong….",ex)
+
+    return
+
+# E-mail Test Dummy Function
+# def send_email(email_from, email_subject, email_body):
+#     print("Email sent:")
+#     print(f"From: {email_from}")
+#     print(f"Subject: {email_subject}")
+#     print(f"Body: {email_body}")
+
+def find_ssl(url, testing=False, test_result=None):
+    if testing:
+        if test_result is None:
+            raise ValueError("test_result argument must be provided in testing mode.")
+        result = test_result
+    else:
+        # Construct the command
+        command = f'echo {url} | openssl s_client -connect {url}:443 -servername {url} | openssl x509 -noout -dates'
+
+        try:
+            # Run the openssl command and capture the output
+            result = subprocess.run(command, shell=True, capture_output=True, text=True, check=True)
+        except subprocess.CalledProcessError as e:
```

no need to catch, let it crash.